onnx/defs/math/defs.cc

```diff
@@ -2178,17 +2178,18 @@ void einsumRankInference(
 
   // Parse the left-hand side
   std::stringstream str(left_equation);
-  while (std::getline(str, term, ',')) {
+  while (!str.eof()) {
```

Comment:
I am trying to understand why the earlier version exited the while-loop at all. Anyone know? The getline spec says that it returns a reference to a stream. (Just want to double-check what the difference in behavior here.)

Reply:
Ok, I understand ... it is implicitly converting the stream object into a bool, after the "getline()" is done ... and so it wasn't processing the "empty-string" getline returns in the last iteration.

Reply:
Thanks for clarifying this